lightning/src/ln/onion_payment.rs

```diff
@@ -601,19 +601,18 @@ where
 
 pub(super) fn check_incoming_htlc_cltv(
 	cur_height: u32, outgoing_cltv_value: u32, cltv_expiry: u32
-) -> Result<(), (&'static str, LocalHTLCFailureReason)> {
+) -> Result<(), LocalHTLCFailureReason> {
```

Comment:
In `decode_incoming_update_add_htlc_onion`, `encode_malformed_error` could have its `message` param dropped perhaps.

Reply:
We lose some information from the `encode_relay_error` calls eg: "Underflow calculating outbound amount or cltv value for blinded forward" just turns into `InvalidOnionBlinding`.

Tension between internal/api struct shows up there.

Reply:
We could add that, but IMO it's not interesting to surface that on the API, and they all map to the same bolt04 code.

So we'd be adding a new enum variant just for the sake of getting rid of a string - not worth the change IMO.

Reply:
It isn't completely clear to me why something like `Underflow calculating outbound amount or cltv value for blinded forward` isn't interesting to surface on the API, while other internal reasons that are also not uniquely mapped to a bolt code are.

But fine to leave as is.

Reply:
To me, blinding errors are _totally_ useless to a forwarding node because somebody else made a mistake.

Other non-bolt04 errors like liquidity errors (actionable) and private channel aliases (interesting, somebody is probing you) just seem much more obvious candidates to be surfaced. 

That said, there's also things like `ZeroAmount` that we've surfaced which arguably aren't interesting either (buggy sender), but they happened to be co-located with other interesting errors - so acknowledge that there isn't one pure logical reason for this, some of it is just what was easy/reasonable to do.

Reply:
Noted in https://github.com/lightningdevkit/rust-lightning/issues/3753.